src/DataFormatter/JSONDataFormatter.php

```diff
@@ -190,4 +191,15 @@ public function convertStringToArray($strData)
     {
         return Convert::json2array($strData);
     }
+
+    public static function cast(FieldType\DBField $dbfield)
+    {
+        if (is_null($dbfield->RAW())) return null;
+        switch (true) {
+            case $dbfield instanceof FieldType\DBInt: return (int)$dbfield->RAW();
+            case $dbfield instanceof FieldType\DBFloat: return (float)$dbfield->RAW();
+            case $dbfield instanceof FieldType\DBBoolean: return (bool)$dbfield->RAW();
+        }
+        return $dbfield->RAW();
+    }
```

Comment:
This method will need to be updated to PSR-2.

I feel like this is circumventing the way the framework should be casting values by default.

Can we not just use `$fieldValue = $obj->obj($fieldName)->JSON()` ?

Reply:
Type checking and casting in the framework would definitely be the better solution, but that would be a major effort.

`$fieldValue = $obj->obj($fieldName)->JSON()` doesn't work because it returns a json formatted string, which would be passed to json_encode again further down the stack, with the result being double encoded strings and arrays, e.g. {"ID":1,"Name":"\"Robbie\""}